python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/_with_span.py

```diff
@@ -67,7 +72,10 @@ def finish_tracing(
                 if value is None:
                     continue
                 try:
-                    self._span.set_attribute(key, value)
+                    if isinstance(value, dict):
+                        self._span.set_attribute(key, json.dumps(value))
+                    else:
+                        self._span.set_attribute(key, value)
```

Comment:
What case is this meant to handle? A bit of a code smell.

Reply:
metadata, it's a dictionary when reading from the context and it fails if we don't convert it to JSON

Reply:
Yeah I can do that instead, good call out